src/receivers/SocketModeReceiver.spec.ts

```diff
@@ -0,0 +1,288 @@
+      const fakeReq = {
+        url: '/hiya',
+      };
+      const fakeRes = {
+        writeHead: sinon.fake(),
+        end: sinon.fake(),
+      };
+      /* eslint-disable-next-line @typescript-eslint/await-thenable */
+      await listener(fakeReq, fakeRes);
+      assert(installProviderStub.generateInstallUrl.calledWith(sinon.match({ metadata, scopes, userScopes })));
+      assert(fakeRes.writeHead.calledWith(200, sinon.match.object));
+      assert(fakeRes.end.called);
+    });
+    it('should return a 404 if a request comes into neither the install path nor the redirect URI path', async () => {
+      // Arrange
+      const fakeServer = new FakeServer();
+      let listener = (_req: any, _res: any) => {};
+      const fakeCreateServer = sinon.fake((handler: (req: any, res: any) => void) => {
+        listener = handler;
+        return fakeServer;
+      });
+      const installProviderStub = sinon.createStubInstance(InstallProvider);
+      const overrides = mergeOverrides(
+        withHttpCreateServer(fakeCreateServer),
+        withHttpsCreateServer(sinon.fake.throws('Should not be used.')),
+      );
+      const SocketModeReceiver = await importSocketModeReceiver(overrides);
+
+      const metadata = 'this is bat country';
+      const scopes = ['channels:read'];
+      const userScopes = ['chat:write'];
+      const receiver = new SocketModeReceiver({
+        appToken: 'my-secret',
+        logger: noopLogger,
+        clientId: 'my-clientId',
+        clientSecret: 'my-client-secret',
+        stateSecret: 'state-secret',
+        scopes,
+        installerOptions: {
+          authVersion: 'v2',
+          installPath: '/hiya',
+          redirectUriPath: '/heyo',
+          metadata,
+          userScopes,
+        },
+      });
+      assert.isNotNull(receiver);
+      receiver.installer = installProviderStub as unknown as InstallProvider;
+      assert.isNotNull(receiver.installer);
+      const fakeReq = {
+        url: '/nope',
+      };
+      const fakeRes = {
+        writeHead: sinon.fake(),
+        end: sinon.fake(),
+      };
+      /* eslint-disable-next-line @typescript-eslint/await-thenable */
+      await listener(fakeReq, fakeRes);
+      assert(fakeRes.writeHead.calledWith(404, sinon.match.object));
+      assert(fakeRes.end.calledWith(sinon.match("route /nope doesn't exist!")));
+    });
+  });
+});
+
+/* Testing Harness */
+
+// Loading the system under test using overrides
+async function importSocketModeReceiver(
+  overrides: Override = {},
+): Promise<typeof import('./SocketModeReceiver').default> {
+  return (await rewiremock.module(() => import('./SocketModeReceiver'), overrides)).default;
+}
+
+// Composable overrides
+function withHttpCreateServer(spy: SinonSpy): Override {
+  return {
+    http: {
+      createServer: spy,
+    },
+  };
+}
+
+function withHttpsCreateServer(spy: SinonSpy): Override {
+  return {
+    https: {
+      createServer: spy,
+    },
+  };
+}
+
+// Fakes
```

Comment:
All of these helper methods / classes down at the bottom were shamelessly copy-pasted from `ExpressReceiver.spec.ts`. If it would be desirable to factor this out and not copy it around, let me know, happy to do it!

Reply:
I think duplicating these few lines of code is okay for now. If we add more functionalities, we can revisit!